src/v/kafka/server/handlers/produce.cc

```diff
@@ -332,6 +333,13 @@ static partition_produce_stages produce_topic_partition(
  */
 static topic_produce_stages
 produce_topic(produce_ctx& octx, produce_request::topic& topic) {
+    /// If topic configuration does not permit, cannot produce onto this topic
+    const model::topic_namespace_view tns_view{
+      model::kafka_namespace, topic.name};
+    const auto topic_cfg = octx.rctx.metadata_cache().get_topic_cfg(tns_view);
+    const bool materialized = topic_cfg ? is_topic_materialized(*topic_cfg)
+                                        : false;
```

Comment:
if there is no topic configuration should we throw an error?

Reply:
Later on in this method there is a query for the topic + partition. If it fails `unknown_topic_or_partition` will be called.